File: php/34-find-first-and-last-position-of-element-in-sorted-array.php
<?php
class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function searchRange($nums, $target) {
        $left = 0;
        $right = count($nums) - 1;
        while ($left <= $right) {
            $middle = ($left+$right) >> 1;
            $mValue = $nums[$middle];
            if ($mValue == $target) {
                $goLeft = $middle;
                while ($nums[$goLeft - 1] === $target) {
                    $goLeft--;
                }
                $pos[] = $goLeft;
                while ($nums[$middle + 1] === $target) {
                    $middle++;
                }
                $pos[] = $middle;
                return $pos;
            } else if ($mValue < $target) {
                $left = $middle + 1;
            } else {
                $right = $middle - 1;
            }
        }
        return [-1, -1];
    }
}

File: go/34-find-first-and-last-position-of-element-in-sorted-array.go
package main

func searchRange(nums []int, target int) []int {
	left := 0
	right := len(nums) - 1
	pos := []int{-1, -1}
	for left <= right {
		middle := (left + right) >> 1
		if nums[middle] == target {
			searchLeft := middle
			for searchLeft-1 >= 0 && nums[searchLeft-1] == target {
				searchLeft--
			}
			pos[0] = searchLeft
			searchRight := middle
			for searchRight+1 <= right && nums[searchRight+1] == target {
				searchRight++
			}
			pos[1] = searchRight
			return pos
		} else if nums[middle] < target {
			left = middle + 1
		} else {
			right = middle - 1
		}
	}
	return pos
}
